libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2131,6 +2131,51 @@ string YulUtilFunctions::revertReasonIfDebug(string const& _message)
 	return revertReasonIfDebug(m_revertStrings, _message);
 }
 
+std::string YulUtilFunctions::decodeReturnParametersFunction(
+	RevertStrings _revertStrings,
+	TypePointers _returnTypes,
+	bool _dynamicReturnSize,
+	string const& _retVars
+)
+{
+	ABIFunctions abi(m_evmVersion, _revertStrings, m_functionCollector);
+
+	string const abiDecode = abi.tupleDecoder(_returnTypes, true);
+	string const functionName =
+		"decode_return_parameters_" +
+		revertStringsToString(_revertStrings) + "_" +
+		abiDecode;
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return util::Whiskers(R"(
+			function <functionName>(pos, returnSize) <?hasRetVars> -> <retVars> </hasRetVars> {
+				<?dynamicReturnSize>
+					// copy dynamic return data out
+					returndatacopy(pos, 0, returndatasize())
+				</dynamicReturnSize>
+
+				// update freeMemoryPointer according to dynamic return size
+				mstore(<freeMemoryPointer>, add(pos, and(add(returnSize, 0x15), not(0x1f))))
+
+				// decode return parameters from external try-call into retVars
+				<?hasRetVars>
+					<retVars> := <abiDecode>(pos, add(pos, returnSize))
+				<!hasRetVars>
+					// retVars: <retVars>
```

Comment:
Shouldn't this always be empty?

Reply:
that's been a debug-print style comment for me when I was hitting a bug. I forgot to remove that comment. sorry.